dist/index.html

```diff
@@ -42,11 +45,62 @@
     <script type="text/javascript" src="itkVtkViewer.js">
     </script>
     <script>
-      var container = document.querySelector('.content');
-      var viewer = itkVtkViewer.processURLParameters(container, { fullscreen: true });
+     async function setupViewerForImJoy(container) {
+      const imjoyRPC = await imjoyLoader.loadImJoyRPC({
+        api_version: '0.2.3'
+      })
+      const api = await imjoyRPC.setupRPC({
+        name: 'itk-vtk-viewer',
+        version: '9.23.x',
+        description: '2D / 3D web image, mesh, and point set viewer using itk.js and vtk.js ',
+        type: 'rpc-window',
+      })
+      api.registerCodec({
+        name: 'itkimage',
+        decoder: itkVtkViewer.utils.convertToItkImage
+      })
+      api.registerCodec({
+        name: 'ndarray',
+        decoder: itkVtkViewer.utils.ndarrayToItkImage
+      })
+      api.export({
+        setup() {
+          api.log("itk-vtk-viewer loaded successfully.")
+        },
+        async run(ctx) {
+          if (ctx.data && ctx.data.image_array) {
+            await this.imshow(ctx.data.image_array)
+          } else {
+            itkVtkViewer.createViewerFromLocalFiles(container)
+          }
+        },
+        async imshow(image_array) {
+          const vtkImage = itkVtkViewer.utils.vtkITKHelper.convertItkToVtkImage(image_array)
+          const dims = vtkImage.getDimensions()
+          const is2D = dims.length === 2 || (dims.length === 3 && dims[2] === 1)
+          itkVtkViewer.createViewer(container, {
+            image: vtkImage,
+            pointSets: null,
```

Comment:
Similar to `imshow`, would be nice to add api for `pointSets` and `geometries`, what do you think?

Reply:
Yes, I'll add the `pointSets` and `geometries` API functions after we have the layers functionality implemented :-).

Reply:
Great! You mean you will eventually support layers from the viewer?
Do you want to also change `imshow` to something else then? to make it easier to be consistent with `showPointSets` later.

Should we decide 2D or 3D for the user? `dims.length === 2 || (dims.length === 3 && dims[2] === 1)` might not be the best choice.

Reply:
> Do you want to also change imshow to something else then? to make it easier to be consistent with showPointSets later.

Yes, we can use `addImage` to go with `addGeometry`, `addPointSet`.

> Should we decide 2D or 3D for the user? dims.length === 2 || (dims.length === 3 && dims[2] === 1) might not be the best choice.

Yes, there is ambiguity with `ndarray` -- `itk/Image` provides an opportunity to explicitly define the dimension, so we will support that, too.

Reply:
Sounds good!

So I guess the idea is if we call `addImage` twice, will we get two layers, correct?

For now, would be nice to have a way to update the image without recreate the entire viewer (e.g. `updateImage` or `setImage`), to support applications such as a live view of the microscope: [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/imjoy-team/micro-manager-imjoy/master?filepath=Micro-Manager-ImJoy-Tutorial.ipynb)

Reply:
> So I guess the idea is if we call addImage twice, will we get two layers, correct?

Yes. At least in the near future, we can only support a single image, so other layers will be made invisible. We can support multiple geometries and point sets.

> For now, would be nice to have a way to update the image without recreate the entire viewer (e.g. updateImage or setImage), to support applications such as a live view of the microscope: ￼

:heart: !

Cool, we will add a method like that, too :-)